bazel/emscripten_toolchain/link_wrapper.py

```diff
@@ -54,8 +43,41 @@
 if rtn != 0:
   sys.exit(1)
 
-js_name = os.path.basename(js_output)
-base_name = os.path.splitext(js_name)[0]
+# Parse the arguments that we gave to the linker to determine what the output
+# file is named and what the output format is.
+parser = argparse.ArgumentParser(add_help=False)
+parser.add_argument('-o')
+parser.add_argument('--oformat')
+(options, _) = parser.parse_known_args(param_file_args)
+output_file = options.o
+oformat = options.oformat
+outdir = os.path.dirname(output_file)
+base_name = os.path.basename(output_file)
+
+# The output file name is the name of the build rule that was built.
+# Add an appropriate file extension based on --oformat.
+if oformat is not None:
```

Comment:
If oformat is None don't we want to default to "js" since that is what emcc will do?

How about `parser.add_argument('--oformat', default='js')` above and then you can remove this condition?